x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -368,6 +370,7 @@
                 .put("enrich_user", new RoleDescriptor("enrich_user", new String[]{ "manage_enrich", "manage_ingest_pipelines", "monitor" },
                         new RoleDescriptor.IndicesPrivileges[]{  RoleDescriptor.IndicesPrivileges.builder()
                             .indices(".enrich-*")
+                            .allowRestrictedIndices(true)
```

Comment:
I am not sure about this one.  The other externally visible built-in roles that are now being given access to restricted indices are where it's some process owned by Elastic but not Elasticsearch itself that's accessing the system index.

But in the case of enrich I would have though it should be like ML where users call enrich APIs and then the enrich plugin accesses system its system indices to perform the implementation details as the `_xpack` user.  Maybe that's not how it works today, but if so that seems like a code smell where internal information is being unnecessarily exposed.

Reply:
You are right @droberts195 users should not get access to `.enrich-*` indices.
Like in all the other cases (ml, transform, watcher, security, etc) APIs should use special internal "origin" access privilege to access the system index and not require the user have that privilege.

I will raise a bug issue.
In scope of this PR, however, I think this role definition is correct because unfortunately the `.enrich-*` namespace is registered as system.

Reply:
Raised https://github.com/elastic/elasticsearch/pull/76461